### ISSUE TITLE
Custom sort key for dataframe.read_parquet with multiple paths as input

### ISSUE THREAD

**Issue by alexgorban** [dask/dask#10045]
[dask.dataframe.read_parquet](https://docs.dask.org/en/stable/generated/dask.dataframe.read_parquet.html) accepts a list of paths as input. It uses [natural_sort_key](https://pydocs.github.io/p/dask/2021.10.0/api/dask.utils.natural_sort_key) under the hood to sort the paths in several places. In my case parquet files are not in the natural order and I'd like to sort them outside and keep the same order inside the read_parquet, so I do monkey patching like this to make it work:

```
with (mock.patch.object(dask.dataframe.io.parquet.core, 'natural_sort_key', new=lambda p: p),
      mock.patch.object(dask.dataframe.io.parquet.utils, 'natural_sort_key', new=lambda p: p)):    
    return dd.read_parquet(paths, index='index', calculate_divisions=True)
```

It would be great if there was a parameter to control the sort key (disable natural_sort_key or provide my own).

**Comment by jrbourbeau** [comment]
Thanks for the issue @alexgorban. Out of curiosity, how are your files being produced? 

@rjzamora does it make sense to use the given file order if users provide a list of paths?

**Comment by rjzamora** [comment]
> does it make sense to use the given file order if users provide a list of paths?

Yes. This has come up before (I’ll need to find the original issue and link later). It may be a bit easier to make it optional to sort a user-provided path list now.

**Comment by alexgorban** [comment]
> Out of curiosity, how are your files being produced?

Files are created using Apache Arrow with filenames sorted alphabetically (pyarrow.Table.sort_by doesn't support custom key functions) and have the following format: `{hash}_{a}_{b}_{c}_{d}.parquet`, ex. `10023947602400723454_1120_000_1140_000`. Their order is important because they have a sorted index column and their metadata stored with this order.  Full details about the dataset format can be found here https://github.com/waymo-research/waymo-open-dataset/blob/master/tutorial/tutorial_v2.ipynb.

Thanks a lot for considering implementing this feature!

**Comment by rjzamora** [comment]
Thanks for explaining [alexgorban](https://github.com/alexgorban)! Please feel free to provide feedback on the proposed solution in https://github.com/dask/dask/pull/10061. The current plan is to simply preserve the order of a user-provided path list, but to use natural sorting when a directory or glob-pattern is specified.
